We will have another simple example on two sample t test (pooled- when the variances are equal). But this time this is a one sided t-test
In a packing plant, a machine packs cartons with jars. It is supposed that a new machine will pack faster on the average than the machine currently used. To test that hypothesis, the times it takes each machine to pack ten cartons are recorded. The results, in seconds, are shown in the tables in the file files_for_lab/machine.txt. Assume that there is sufficient evidence to conduct the t test, does the data provide sufficient evidence to show if one machine is better than the other

In [ ]:
# Let's start by calculating the test statistic using the provided data for the new and old machines:

In [1]:
# New machine times (in seconds):
X_new =[42.1,41,41.3,41.8,42.4,42.8,43.2,42.3,41.8,42.7]

# Old machine times (in seconds):

X_old=[42.7,43.6,43.8,43.3,42.5,43.5,43.1,41.7,44,44.1]

Let's calculate the mean, standard deviation, and sample size for each machine. Then, we'll proceed to calculate the test statistic.

In [ ]:
# Let's start by calculating the mean, standard deviation, and sample size for each machine:

In [2]:
import numpy as np

times_new = np.array([42.1, 41, 41.3, 41.8, 42.4, 42.8, 43.2, 42.3, 41.8, 42.7])

mean_new = np.mean(times_new)
std_new = np.std(times_new, ddof=1)
size_new = len(times_new)

print("New Machine:")
print("Mean:", mean_new)
print("Standard Deviation:", std_new)
print("Sample Size:", size_new)

New Machine:
Mean: 42.14
Standard Deviation: 0.6834552736727638
Sample Size: 10


In [3]:
times_old = np.array([42.7, 43.6, 43.8, 43.3, 42.5, 43.5, 43.1, 41.7, 44, 44.1])

mean_old = np.mean(times_old)
std_old = np.std(times_old, ddof=1)
size_old = len(times_old)

print("\nOld Machine:")
print("Mean:", mean_old)
print("Standard Deviation:", std_old)
print("Sample Size:", size_old)


Old Machine:
Mean: 43.230000000000004
Standard Deviation: 0.7498888806572157
Sample Size: 10


We will use the two-sample t-test to calculate the test statistic and the p-value. If the p-value is less than the significance level (α), we will reject the null hypothesis and conclude that there is sufficient evidence to show that one machine is better than the other. Otherwise, we will fail to reject the null hypothesis.

In [4]:
from scipy import stats

t_statistic, p_value = stats.ttest_ind(times_new, times_old)

print("Test Statistic:", t_statistic)
print("P-value:", p_value)

Test Statistic: -3.3972307061176026
P-value: 0.0032111425007745158


In [6]:
alpha = 0.05
if p_value < alpha:
    print("Reject the null hypothesis. There is sufficient evidence to show that one machine is better than the other.")
else:
    print("Fail to reject the null hypothesis. There is not sufficient evidence to show that one machine is better than the other.")

Reject the null hypothesis. There is sufficient evidence to show that one machine is better than the other.


An additional problem (not mandatory): In this case we can't assume that the population variances are equal. Hence in this case we cannot pool the variances. Independent random samples of 17 sophomores and 13 juniors attending a large university yield the following data on grade point averages. Data is provided in the file files_for_lab/student_gpa.txt. At the 5% significance level, do the data provide sufficient evidence to conclude that the mean GPAs of sophomores and juniors at the university differ?

Test statistics can be calculated as: link to the image - Test statistics calculation for Unpooled Variance Case

Degrees of freedom is (n1-1)+(n2-1).

In [7]:
from scipy.stats import ttest_ind

sophomores_gpa = [3.04, 1.71, 3.3, 2.88, 2.11, 2.6, 2.92, 3.6, 2.28, 2.82, 3.03, 3.13, 2.86, 3.49, 3.11, 2.13, 3.27]
juniors_gpa = [2.56, 2.77, 2.7, 3, 2.98, 3.47, 3.26, 3.2, 3.19, 2.65, 3, 3.39, 2.58]

t_stat, p_value = ttest_ind(sophomores_gpa, juniors_gpa, equal_var=False)
print("Test Statistic:", t_stat)
print("P-value:", p_value)

Test Statistic: -0.9231495630900278
P-value: 0.3642180675348571


In [8]:
alpha = 0.05
if p_value < alpha:
    print("Reject the null hypothesis")
else:
    print("fail to reject the null hypothesis")

fail to reject the null hypothesis


With a significance level of 0.05, since the p-value (0.3642) is greater than 0.05, we fail to reject the null hypothesis. This means that there is not enough evidence to conclude that the mean GPAs of sophomores and juniors at the university differ.